frontend/src/components/student/PointsPage/Points.js

```diff
@@ -22,23 +20,54 @@ export default function Points() {
       .catch(() => {
         setPointsData(null)
       })
+    StudentService.getTotalReceivedPoints()
+      .then((response) => {
+        setTotalPointsData(response)
+      })
+      .catch(() => {
+        setTotalPointsData(null)
+      })
   }, [])
 
   return (
     <Content>
       <Row className='m-0'>
         <Col className='p-0'>
-          <PercentageCircle percentageValue={percentageValue} points={points} maxPoints={maxPoints} />
+          {totalPointsData === undefined ? (
+            <Spinner animation={'border'} />
+          ) : totalPointsData == null ? (
+            <p>Błąd</p>
+          ) : (
+            <PercentageCircle
+              percentageValue={Math.round(
+                100 * (totalPointsData.totalPointsReceived / totalPointsData.totalPointsPossibleToReceive)
+              )}
+              points={totalPointsData.totalPointsReceived}
+              maxPoints={totalPointsData.totalPointsPossibleToReceive}
+            />
+          )}
         </Col>
         <Col className='p-0 justify-content-center d-flex flex-column'>
+          {totalPointsData === undefined ? (
+            <Spinner animation={'border'} />
+          ) : totalPointsData == null ? (
+            <p>Błąd</p>
+          ) : (
+            <>
+              <h5>
+                <strong>Twój wynik to: {totalPointsData.totalPointsReceived}pkt</strong>
+              </h5>
+              <h5>
+                <strong>
+                  Co stanowi:{' '}
+                  {100 * (totalPointsData.totalPointsReceived / totalPointsData.totalPointsPossibleToReceive)}%
+                </strong>
+              </h5>
+            </>
+          )}
```

Comment:
Niepotrzebne powtarzanie kodu. Wyciągnij sprawdzenie null / undefined bezpośrednio pod Row zbierający kolumny, a jeżeli nie jest ani undefined ani null to wtedy te dwie Col. Trochę mniej zduplikowanego kodu ;)